[4972] Improve Search behavior in Browse Components tool of Page Builder

aachavarria opened the pull request:
BrowseComponentsPanel: Moving styles and translation to his own file.
Updates on BrowseComponents Panel to use all contentTypes filter.
fetchItemsByContentType service: fixing service to return only components of the selected contentType;

https://github.com/craftercms/craftercms/issues/4972

aachavarria committed:
BrowseComponentsPanel: Moving styles and translation to his own file.
Updates on BrowseComponents Panel to use all contentTypes filter.
fetchItemsByContentType service: fixing service to return only components of the selected contentType;

rart commented on ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx:
```diff
@@ -179,7 +87,10 @@ export default function PreviewBrowseComponentsPanel() {
-            .filter((item: ContentInstance) => item.craftercms.contentTypeId === source.contentTypeFilter) || [];
+            .filter(
+              (item: ContentInstance) =>
+                source.contentTypeFilter === 'all' || item.craftercms.contentTypeId === source.contentTypeFilter
```
- too inefficient to iterate through the whole array when you know in advanced that you needn't apply the filter at all
- also, the `||` is misplaced; it doesn't avoid an exception if the .map returns undefined
```js
let items = source.page[source.pageNumber]?.map(...) ?? []
if (source.contenTypeFilter !== 'all') 
  items = items.filter(...)

rart commented on ui/app/src/state/epics/contentTypes.ts:
```diff
@@ -28,29 +28,30 @@ import { fetchItemsByContentType } from '../../services/content';
-  );
-
-export default [fetch, fetchComponentsByContentType] as Epic[];
+] as Epic[];
```
```suggestion
] as CrafterCMSEpic[];
```

rart commented on ui/app/src/state/actions/preview.ts:
```diff
@@ -247,15 +247,9 @@ export const fetchAssetsPanelItemsComplete = /*#__PURE__*/ createAction<SearchRe
-  };
-}
+export const fetchComponentsByContentType = /*#__PURE__*/ createAction<Partial<ComponentsContentTypeParams>>(
+  FETCH_COMPONENTS_BY_CONTENT_TYPE
```
we could/should discard these constants

rart commented on ui/app/src/services/content.ts:
```diff
@@ -462,31 +462,37 @@ export function fetchItemsByContentType(
   }).pipe(
     map<AjaxResponse, { count: number; paths: string[] }>(({ response }) => ({
       count: response.result.total,
-      paths: response.result.items.filter((item) => item.type === options.type).map((item) => item.path)
+      paths: response.result.items.map((item) => item.path)
```
Looks like you're dropping support for the type filter. why?
if we do it, then you need to drop it from the options of this method too.

aachavarria replied:
I was doing some testing and the result from the fetch is only including components, so looks like the filter is not doing anything, i think in the past the fetch result included groovy or something similar.

rart commented on ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx:
```diff
@@ -296,60 +201,46 @@ interface BrowsePanelUIProps {
 
 function BrowsePanelUI(props: BrowsePanelUIProps) {
-  const { componentsResource, classes, onPageChanged, onDragStart, onDragEnd, chooseContentTypeImageUrl } = props;
+  const { componentsResource, classes, onPageChanged, onDragStart, onDragEnd } = props;
```
Let's break the individual components into files

rart commented on ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx:
```diff
@@ -296,60 +201,46 @@ interface BrowsePanelUIProps {
+        <EmptyState
+          title={formatMessage(translations.noResults)}
+          classes={{ image: classes.noResultsImage, title: classes.noResultsTitle }}
         />
```
When there aren't any results, is there an empty message elsewhere? 
I'd also like to add a note, that only shared components are shown here. Let's discuss.

aachavarria replied:
there is an  `*<EmptyState/>`, let's discuss tomorrow

aachavarria committed:
PreviewBrowseComponentsPanel.tsx: removing unnecessary imports;
Creating own component for PreviewBrowseComponentsPanelUI.tsx;
Improving resource resultSelector logic;
Removing type prop from ComponentsContentTypeParams;

aachavarria committed:
removing level-descriptor from contentTypes and components;
adding helperText "Only shared components are shown here";

aachavarria committed:
updating styles for helperTextWrapper
